### PR TITLE
[WIP] self-gravity with subcycling

### PR DESCRIPTION
**Pull request by BenWibking**
Work in progress.

### CODE REVIEW

File: src/Gravity_impl.hpp [hunk 1]
```diff
@@ -0,0 +1,1069 @@
...
+    gmv.push_back(sim->Geom(ilev + crse_level));
+    bav.push_back(rhs[ilev]->boxArray());
+    dmv.push_back(rhs[ilev]->DistributionMap());
+  }
+
+  LPInfo info;
+  info.setAgglomeration(gravity::mlmg_agglomeration != 0);
+  info.setConsolidation(gravity::mlmg_consolidation != 0);
+
+  MLPoisson mlpoisson(gmv, bav, dmv, info);
```
Comment (BenWibking) [inline src/Gravity_impl.hpp:1031]: Fails here

**Comment by psharda** [comment]
@BenWibking this PR shouldn't be too hard to complete now right?

**Comment by BenWibking** [comment]
> @BenWibking this PR shouldn't be too hard to complete now right?

You're more than welcome to finish the implementation :)

**Comment by psharda** [comment]
Continued in #499

**Pull request closed without merging**